clang/lib/3C/ProgramInfo.cpp

```diff
@@ -327,37 +327,52 @@ bool ProgramInfo::link() {
 
   // For every global function that is an unresolved external, constrain 
   // its parameter types to be wild. Unless it has a bounds-safe annotation. 
-  for (const auto &U : ExternFunctions) {
+  for (const auto &U : ExternalFunctionFVCons) {
+    std::string FuncName = U.first;
+    FVConstraint *G = U.second;
     // If we've seen this symbol, but never seen a body for it, constrain
     // everything about it.
-    std::string FuncName = U.first;
-    if (!U.second && !isExternOkay(FuncName)) {
-      // Some global symbols we don't need to constrain to wild, like 
-      // malloc and free. Check those here and skip if we find them. 
-      FVConstraint *G = getExtFuncDefnConstraint(FuncName);
-      assert("Function constraints could not be found!" && G != nullptr);
+    // Some global symbols we don't need to constrain to wild, like
+    // malloc and free. Check those here and skip if we find them.
+    if (!G->hasBody() && !isExternOkay(FuncName)) {
 
       // If there was a checked type on a variable in the input program, it
       // should stay that way. Otherwise, we shouldn't be adding a checked type
       // to an extern function.
       std::string Rsn =
-        "Unchecked pointer in parameter or return of external function " +
-        FuncName;
+          "Unchecked pointer in parameter or return of external function " +
+          FuncName;
       if (!G->getReturnVar()->getIsGeneric())
         G->getReturnVar()->constrainToWild(CS, Rsn);
       for (unsigned I = 0; I < G->numParams(); I++)
         if (!G->getParamVar(I)->getIsGeneric())
           G->getParamVar(I)->constrainToWild(CS, Rsn);
     }
   }
+  // repeat for static functions
+  for (const auto &U :StaticFunctionFVCons) {
+    for (const auto &V :U.second) {
+
+      std::string FileName = U.first;
+      std::string FuncName = V.first;
+      FVConstraint *G = V.second;
+      if (!G->hasBody() && !isExternOkay(FuncName)) {
```

Comment:
Why would you do `isExternOkay` for a `static`? By its nature, a `static` cannot be `extern` so it should never be OK for the reason an `extern` (like `malloc`) would be. I would drop this clause. 

I guess a related point is what it means to "have a body." Does `G->hasBody()` do the same thing for
```
static void foo(int *x);
```
as for
```
static void foo(int *x) { }
```
? I think the error we are having is for the latter case?

Finally, this situation is weird because it's broken C code. If you have a `static` with no definition, it will not link properly. So 3C is just being "forgiving" here. I think it's worth adding a comment pointing this out, to avoid confusion.